protocol/communities/community_event.go

```diff
@@ -2,413 +2,257 @@ package communities
-				return err
-			}
-			o.unbanUserFromCommunity(pk)
+		if len(e.MemberToAction) == 0 {
+			return errors.New("invalid community member unban event")
 		}
+
 	case protobuf.CommunityEvent_COMMUNITY_TOKEN_ADD:
-		o.config.CommunityDescription.CommunityTokensMetadata = append(o.config.CommunityDescription.CommunityTokensMetadata, communityEvent.TokenMetadata)
+		if e.TokenMetadata == nil || len(e.TokenMetadata.ContractAddresses) == 0 {
+			return errors.New("invalid add community token event")
+		}
 	}
 	return nil
 }
 
-func (o *Community) NewCommunityEventClock() uint64 {
-	return uint64(time.Now().Unix())
-}
-
-func (o *Community) addNewCommunityEvent(event *CommunityEvent) error {
-	err := validateCommunityEvent(event)
-	if err != nil {
-		return err
-	}
-
-	// All events must be built on top of the control node CommunityDescription
-	// If there were no events before, extract CommunityDescription from CommunityDescriptionProtocolMessage
-	// and check the signature
-	if o.config.EventsData == nil || len(o.config.EventsData.EventsBaseCommunityDescription) == 0 {
-		_, err := validateAndGetEventsMessageCommunityDescription(o.config.CommunityDescriptionProtocolMessage, o.ControlNode())
-		if err != nil {
-			return err
-		}
-
-		o.config.EventsData = &EventsData{
-			EventsBaseCommunityDescription: o.config.CommunityDescriptionProtocolMessage,
-			Events:                         []CommunityEvent{},
-		}
-	}
+// Events with the same PropertyTypeID affects the same Community property.
+func (e *CommunityEvent) PropertyTypeID() string {
+	switch e.Type {
+	case protobuf.CommunityEvent_COMMUNITY_EDIT:
+		return "COMMUNITY_EDIT"
+
+	case protobuf.CommunityEvent_COMMUNITY_MEMBER_TOKEN_PERMISSION_CHANGE,
+		protobuf.CommunityEvent_COMMUNITY_MEMBER_TOKEN_PERMISSION_DELETE:
+		return fmt.Sprintf("TOKEN_PERMISSION-%s", e.TokenPermission.Id)
+
+	case protobuf.CommunityEvent_COMMUNITY_CATEGORY_CREATE,
+		protobuf.CommunityEvent_COMMUNITY_CATEGORY_DELETE,
+		protobuf.CommunityEvent_COMMUNITY_CATEGORY_EDIT,
+		protobuf.CommunityEvent_COMMUNITY_CATEGORY_REORDER:
+		return fmt.Sprintf("COMMUNITY_CATEGORY-%s", e.CategoryData.CategoryId)
+
+	case protobuf.CommunityEvent_COMMUNITY_CHANNEL_CREATE,
+		protobuf.CommunityEvent_COMMUNITY_CHANNEL_DELETE,
+		protobuf.CommunityEvent_COMMUNITY_CHANNEL_EDIT,
+		protobuf.CommunityEvent_COMMUNITY_CHANNEL_REORDER:
+		return fmt.Sprintf("COMMUNITY_CHANNEL-%s", e.ChannelData.ChannelId)
+
+	case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_ACCEPT,
+		protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_REJECT,
+		protobuf.CommunityEvent_COMMUNITY_MEMBER_KICK,
+		protobuf.CommunityEvent_COMMUNITY_MEMBER_BAN,
+		protobuf.CommunityEvent_COMMUNITY_MEMBER_UNBAN:
+		return fmt.Sprintf("MEMBER_STATE-%s", e.MemberToAction)
 
-	event.Payload, err = proto.Marshal(event.ToProtobuf())
-	if err != nil {
-		return err
+	case protobuf.CommunityEvent_COMMUNITY_TOKEN_ADD:
+		return fmt.Sprintf("TOKEN_METADATA-%s", e.TokenMetadata.Name)
```

Comment:
Should we put those prefixes as constants? I assume they are reused somewhere else